### PR TITLE
schemaTracker: lazy init table info in schema tracker

### LINKED ISSUE #1260: "fail to initial unit Sync of subtask" with many do tables
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   `start-task`

2. What did you expect to see?

   no errors

3. What did you see instead?

image

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.0
        ```

goroutines:

image

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #1260 

### What is changed and how it works?
create table in tracker when get table first time.

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: syncer/checkpoint.go [hunk 1]
```diff
@@ -914,3 +909,13 @@ func (cp *RemoteCheckPoint) parseMetaData() (*binlog.Location, *binlog.Location,
 
 	return loc, loc2, err
 }
+
+// GetFlushedTableInfo implements CheckPoint.GetFlushedTableInfo
+func (cp *RemoteCheckPoint) GetFlushedTableInfo(schema string, table string) *model.TableInfo {
+	if tables, ok := cp.points[schema]; ok {
```
Comment (csuzhangxc) [inline syncer/checkpoint.go:915]: Is `Lock()` needed here?

File: syncer/checkpoint.go [hunk 2]
```diff
@@ -771,29 +775,20 @@ func (cp *RemoteCheckPoint) Load(tctx *tcontext.Context, schemaTracker *schema.T
...
 
-		var ti model.TableInfo
+		var ti *model.TableInfo
 		if !bytes.Equal(tiBytes, []byte("null")) {
 			// only create table if `table_info` is not `null`.
 			if err = json.Unmarshal(tiBytes, &ti); err != nil {
 				return terror.ErrSchemaTrackerInvalidJSON.Delegate(err, cpSchema, cpTable)
 			}
-
-			if schemaTracker != nil {
```
Comment (csuzhangxc) [inline syncer/checkpoint.go:781]: We can remove `, schemaTracker *schema.Tracker` for `Load` now?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**